File: back/DevC/src/main/java/com/back/devc/domain/member/member/repository/MemberRepository.java
package com.back.devc.domain.member.member.repository;

import com.back.devc.domain.member.member.entity.AuthProvider;
import com.back.devc.domain.member.member.entity.Member;
import com.back.devc.domain.member.member.entity.MemberStatus;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface MemberRepository extends JpaRepository<Member, Long> {

    boolean existsByEmail(String email);

    boolean existsByNickname(String nickname);

    Optional<Member> findByEmail(String email);

    Optional<Member> findByProviderAndProviderUserId(AuthProvider provider, String providerUserId);

    // 닉네임 검색 (부분 일치)
    Page<Member> findByNicknameContainingIgnoreCase(String nickname, Pageable pageable);
    Page<Member> findByStatus(MemberStatus status, Pageable pageable);

    Page<Member> findByNicknameContainingOrEmailContaining(
            String nickname,
            String email,
            Pageable pageable
    );

    Page<Member> findByStatusAndNicknameContainingOrStatusAndEmailContaining(
            MemberStatus status1,
            String nickname,
            MemberStatus status2,
            String email,
            Pageable pageable
    );

    List<Member> findAllByUserIdIn(List<Long> userIds);

}

File: back/DevC/src/main/kotlin/com/back/devc/domain/member/member/repository/MemberRepository.kt
package com.back.devc.domain.member.member.repository

import com.back.devc.domain.member.member.entity.AuthProvider
import com.back.devc.domain.member.member.entity.Member
import com.back.devc.domain.member.member.entity.MemberStatus
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface MemberRepository : JpaRepository<Member, Long> {

    fun existsByEmail(email: String): Boolean

    fun existsByNickname(nickname: String): Boolean

    fun findByEmail(email: String): Optional<Member>

    fun findByProviderAndProviderUserId(
        provider: AuthProvider,
        providerUserId: String
    ): Optional<Member>

    fun findByNicknameContainingIgnoreCase(
        nickname: String,
        pageable: Pageable
    ): Page<Member>

    fun findByStatus(
        status: MemberStatus,
        pageable: Pageable
    ): Page<Member>

    fun findByNicknameContainingOrEmailContaining(
        nickname: String,
        email: String,
        pageable: Pageable
    ): Page<Member>

    fun findByStatusAndNicknameContainingOrStatusAndEmailContaining(
        status1: MemberStatus,
        nickname: String,
        status2: MemberStatus,
        email: String,
        pageable: Pageable
    ): Page<Member>
}